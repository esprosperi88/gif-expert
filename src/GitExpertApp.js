
import React, { useState } from 'react';
import AddCategoria from './components/AddCategoria';
import { GifGrid } from './components/GifGrid';
export const GitExpertApp =({defaultCategorias = ['One Punch']}) =>{

    const [categorias,setCategorias] = useState(defaultCategorias);
    
    const handleAdd = () =>{
        setCategorias([...categorias,'newCategoria']);
        console.log("boton");
    }

    return (<>
                <h2>gif-expert</h2>
                <AddCategoria setCategorias={setCategorias} />
                <hr/>
                <ol>
                    {
                        categorias.map(categoria => (
                         <GifGrid   
                                    key={ categoria }  
                                    categoria= {categoria} 
                               />
                        ))
                    }
                </ol>
                <hr/>
                <button onClick={handleAdd}>button</button>  
            </>
    );

}

export default GitExpertApp;