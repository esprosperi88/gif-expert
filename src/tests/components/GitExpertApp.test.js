import React from "react";
import GitExpertApp from '../../GitExpertApp';
import {shallow}   from 'enzyme'

describe('Prueba component - GitExpertApp ',() => {
    
    let wrapper;
    const  categorias =["dragon ball","goku"];
    beforeEach( () => {   
        jest.clearAllMocks();

        wrapper = shallow(<GitExpertApp defaultCategorias= {categorias} />);
    });

    test('GitExpertApp - snapshot',() => {
        expect (wrapper).toMatchSnapshot();
    });

    test('GitExpertApp - verifico si crea los items de categoria',() => {
        expect (wrapper.find('GifGrid').length).toBe(categorias.length)
    });

});