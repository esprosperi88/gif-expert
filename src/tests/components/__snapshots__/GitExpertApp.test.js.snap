// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba component - GitExpertApp  GitExpertApp - snapshot 1`] = `
<Fragment>
  <h2>
    gif-expert
  </h2>
  <AddCategoria
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="dragon ball"
      key="dragon ball"
    />
    <GifGrid
      categoria="goku"
      key="goku"
    />
  </ol>
  <hr />
  <button
    onClick={[Function]}
  >
    button
  </button>
</Fragment>
`;
